wpilibj/src/main/java/edu/wpi/first/wpilibj/ADIS16470_IMU.java

```diff
@@ -391,16 +392,39 @@ public ADIS16470_IMU(
         return;
       }
 
-      // Set IMU internal decimation to 4 (output data rate of 2000 SPS / (4 + 1) =
-      // 400Hz)
-      writeRegister(DEC_RATE, 4);
+      // Set IMU internal decimation to 4 (output data rate of 2000 SPS / (4 + 1) = 400Hz), output bandwidth = 200Hz
+      if(readRegister(DEC_RATE) != 0x0004){
+        writeRegister(DEC_RATE, 0x0004);
+        m_needs_flash = true;
+        DriverStation.reportWarning("ADIS16470: DEC_RATE register configuration inconsistent! Scheduling flash update.", false);
+      }
+
+      // Set data ready polarity (HIGH = Good Data), Disable gSense Compensation and PoP
+      if(readRegister(MSC_CTRL) != 0x0001){
+        writeRegister(MSC_CTRL, 0x0001); 
+        m_needs_flash = true;
+        DriverStation.reportWarning("ADIS16470: MSC_CTRL register configuration inconsistent! Scheduling flash update.", false);
+      }
 
-      // Set data ready polarity (HIGH = Good Data), Disable gSense Compensation and
-      // PoP
-      writeRegister(MSC_CTRL, 1);
+      // Disable IMU internal Bartlett filter (200Hz bandwidth is sufficient)
+      if(readRegister(FILT_CTRL) != 0x0000{
+        writeRegister(FILT_CTRL, 0x0000);
+        m_needs_flash = true;
+        DriverStation.reportWarning("ADIS16470: FILT_CTRL register configuration inconsistent! Scheduling flash update.", false);
+      }
 
-      // Configure IMU internal Bartlett filter
-      writeRegister(FILT_CTRL, 0);
+      // If any registers on the IMU don't match the config, trigger a flash update
+      if(m_needs_flash){
+        DriverStation.reportWarning("ADIS16470: Register configuration changed! Starting IMU flash update.", false);
+        writeRegister(GLOB_CMD, 0x0008);
+        // Wait long enough for the flash update to finish (72ms minimum as per the datasheet)
+        Timer.delay(0.3);
+        DriverStation.reportWarning("ADIS16470: Flash update finished!", false);
+        m_needs_flash = false;
+      }
+      else {
+        DriverStation.reportWarning("ADIS16470: and RAM configuration consistent. No flash update required!", false);
```

Comment:
Why is this a warning?

Reply:
Because it will always show up in the DS logs. Prints get filtered.